### PR TITLE
Break up and show more progression during an Xcode build

### LINKED ISSUE #14659: Selectively stream xcodebuild output
Building on iOS takes a long time and it's hard to tell if anything's happening during
`Running Xcode build...`

There are interesting feedback we can print in xcode_backend.sh but we don't show any output from xcodebuild unless there's an error. Selectively stream some output from xcode_backend.sh while keeping the rest of the log to show only when there's an error.

### PR DESCRIPTION
**Pull request by xster**
Fixes: #14659

Success build:
```
Launching lib/main.dart on iPhone in debug mode...
Signing iOS app for device deployment using developer identity: "iPhone Developer: Xiao Yu (2L257N75XL)"
Running Xcode clean...                                1.3s
Starting Xcode build...
 ├─Building Dart code...                       3.7s
 ├─Assembling Flutter resources...             6.0s
 └─Compiling, linking and signing...           6.1s
Xcode build done                                     17.4s
Installing and launching...                           9.1s
Syncing files to device iPhone...                     3.1s
```

Failed build:
```
Launching lib/main.dart on iPhone in debug mode...
Signing iOS app for device deployment using developer identity: "iPhone Developer: Xiao Yu (2L257N75XL)"
Running Xcode clean...                                1.2s
Starting Xcode build...
 ├─Building Dart code...                       0.8s
 ├─Assembling Flutter resources...             2.2s
 └─Compiling, linking and signing...           0.2s
Xcode build done                                      5.0s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **

Xcode's output:
↳
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    /Users/xster/development/flutter/examples/flutter_gallery/ios/Runner/AppDelegate.m:7:57: error: use of undeclared identifier 'craps'
      [GeneratedPluginRegistrant registerWithRegistry:self];craps out
                                                            ^
    1 error generated.
Could not build the precompiled application for the device.

Error launching application on iPhone.
```

### CODE REVIEW

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 1]
```diff
@@ -346,14 +348,58 @@ Future<XcodeBuildResult> buildXcodeProject({
...
+
+  if (logger.supportsColor) {
+    scriptOutputPipeTempDirectory = fs.systemTempDirectory
+        .createTempSync('flutter_build_log_pipe');
+    final File scriptOutputPipeFile =
+        scriptOutputPipeTempDirectory.childFile('pipe_to_stdout');
+    os.makePipe(scriptOutputPipeFile.path);
+
+    Future<void> listenToScriptOutputLine() async {
+      final List<String> lines = await scriptOutputPipeFile.readAsLines();
```
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/mac.dart]: Would it be better to use `openRead()` and stream the lines as they come?

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

**Comment by cbracken** [comment]
The existing code needs refactoring for testability. As-is it's pretty untestable :(

**Comment by xster** [comment]
Ya, I definitely want to do a big rewrite after IO #10756

File: packages/flutter_tools/lib/src/ios/mac.dart (hunk 1 above)
Comment (xster) [reply packages/flutter_tools/lib/src/ios/mac.dart]: Discussed offline. 
It doesn't seem to work
https://github.com/dart-lang/sdk/issues/32191

For now, we'll never print more than one from the other end of the pipe so it should be ok.

**Pull request merged**